modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -824,7 +824,13 @@ def from_labels(self) -> "PandasDataframe":
         def from_labels_executor(df, **kwargs):
             # Setting the names here ensures that external and internal metadata always match.
             df.index.names = new_column_names
-            return df.reset_index()
+
+            if any(name_level in df.columns for name_level in df.index.names):
+                columns_to_add = df.index.to_frame().reset_index(drop=True)
+                df = df.reset_index(drop=True)
+                return pandas.concat([columns_to_add, df], axis=1)
```

Comment:
```suggestion
                columns_to_add = df.index.to_frame()
                columns_to_add.reset_index(drop=True, inplace=True)
                df = df.reset_index(drop=True)
                return pandas.concat([columns_to_add, df], axis=1, copy=False)
```
should be a bit faster as we don't create unnecessary copies

Reply:
It would also be good to place a comment before the if-branch of why we ignore columns duplication in the kernel function. It's hard to understand the logic of having these two if-branches out of the context of this PR.

Reply:
> should be a bit faster as we don't create unnecessary copies

Applied with additional changes: ` df = df.reset_index(drop=True)` --> `df.reset_index(drop=True, inplace=True)`.

> It would also be good to place a comment before the if-branch of why we ignore columns duplication in the kernel function. It's hard to understand the logic of having these two if-branches out of the context of this PR.

Agree, the note-comment is added.

Reply:
I feel safe to set `inplace=True` for `columns_to_add` frame as it is a COPY of the `df.index` and so nothing depends on it. I don't feel the same about `df`, since it's the object that is stored in the partition and so modifying it we modify the partition's content, isn't it?

Reply:
Looks reasonable, fixed.